content/posts/cicd_ha.md

```diff
@@ -0,0 +1,132 @@
+
+Regarding the *"agreed level of operational performance"* from our previous HA definition, in manufacturing production lines there is no need to replicate the entire manufacturing platform or production lines to have a highly available service. This means that the lines can afford a stoppage of a couple of minutes with no impact on the cadence of production. From a business and production perspective this is a ***highly available service***.
+
+One of the reasons of not working with replicated *active-active* running production lines as HA strategy is because is not cost effective for the service interruption recovery requirements. In other words, we get the same final results using easy scalable production lines that can recover fast from a failure, without the need of expensive *active-active* infrastructures.
+
+### Resilient vs Highly Available
+
+Once said that, one of the most important features about a *Continuous Delivery* system (note that I am talking in general, not only software) to be highly available, is **resilience**. Understood as the capacity of our system components to recover on time, producing a complete stable, durable and efficient execution that doesn't impact the overall service (e.g. my deployment or release frequency and quality of final artifacts are the same, with or without short downtimes. Just because my platform and systems are *resilient* to failures).
+
+Let's back then to the question:
+
+> ***What platform configuration do I need for a highly available service?***
+
+To answer this question, it's very important to think about **resilience first**, because having a resilient platform and architecture is usually enough to deliver a highly available service, being able to recover on time to not impact the final service. But in some scenarios where a couple of minutes of service blockage means loss of millions of dollars (like Amazon retail previous example), resilient solutions are not enough. In this case then you probably need to work on very restrictive HA replication infrastructures. The cost of replication complexity need to be worth it.
+
+## So CI/CD HA is just Resilience
+
+Coming back to main topic... **YES**, it is about resilient solutions. HA in CI/CD is not about looking for an *active-active* load balancing platform that is able to orchestrate CI/CD pipelines. It is about assuring enough *availability* of my pipelines executions to deliver my software on time and covering all expectations (deployment frequency, failure recovery, lead time, quality standards, etc.).
+
+What does resilient architecture mean in CI/CD pipeline orchestration? I think that following solution architecture concepts are a must in order to offer a resilient *highly available* solution:
+
+* Orchestration layer does not have the same needs that execution layer. Pipeline orchestrators and pipeline executors have different scalability  requirements, and because of that **decoupling execution from orchestration is a must**.
```

Comment:
"Orchestration layer does not have the same needs that execution layer."
to
"Orchestration layer does not have the same needs as the execution layer."